openlibrary/utils/lcc.py

```diff
@@ -0,0 +1,185 @@
+    (?P<letters>[A-HJ-NP-VWZ][A-Z-]{0,2})
+    \s?
+    (?P<number>\d{1,4}(\.\d+)?)?
+    (?P<cutter1>[\s.][^\d\s\[]{1,3}\d*\S*)?
+    (?P<rest>\s.*)?
+    $
+''', re.IGNORECASE | re.X)
+
+
+def short_lcc_to_sortable_lcc(lcc):
+    """
+    See Sorting section of doc above
+    :param str lcc: unformatted lcc
+    :rtype: basestring|None
+    """
+    m = LCC_PARTS_RE.match(clean_raw_lcc(lcc))
+    if not m:
+        return None
+
+    parts = m.groupdict()
+    parts['letters'] = parts['letters'].upper().ljust(3, '-')
+    parts['number'] = float(parts['number'] or 0)
+    parts['cutter1'] = '.' + parts['cutter1'].lstrip(' .') if parts['cutter1'] else ''
+    parts['rest'] = ' ' + parts['rest'].strip() if parts['rest'] else ''
+
+    return '%(letters)s%(number)013.8f%(cutter1)s%(rest)s' % parts
+
+
+def sortable_lcc_to_short_lcc(lcc):
+    """
+    As close to the inverse of make_sortable_lcc as possible
+    :param basestring lcc:
+    :rtype: basestring
+    """
+    m = LCC_PARTS_RE.match(lcc)
+    parts = m.groupdict()
+    parts['letters'] = parts['letters'].strip('-')
+    parts['number'] = parts['number'].strip('0').strip('.')  # Need to do in order!
+    parts['cutter1'] = parts['cutter1'].strip(' ') if parts['cutter1'] else ''
+    parts['rest'] = ' ' + parts['rest'].strip() if parts['rest'] else ''
+
+    return '%(letters)s%(number)s%(cutter1)s%(rest)s' % parts
+
+
+def clean_raw_lcc(raw_lcc):
+    """
+    Remove noise in lcc before matching to LCC_PARTS_RE
+    :param basestring raw_lcc:
+    :rtype: basestring
+    """
+    lcc = collapse_multiple_space(raw_lcc.replace('\\', ' ').strip(' '))
+    if ((lcc.startswith('[') and lcc.endswith(']')) or
+            (lcc.startswith('(') and lcc.endswith(')'))):
+        lcc = lcc[1:-1]
+    return lcc
+
+
+def normalize_lcc_prefix(prefix):
+    """
+    :param str prefix: An LCC prefix
+    :return: Prefix transformed to be a prefix for sortable LCC
+    :rtype: str|None
+    """
+    if re.match(r'^[A-Z]+$', prefix, re.I):
+        return prefix
+    else:
+        # A123* should be normalized to A--0123*
+        # A123.* should be normalized to A--0123.*
+        # A123.C* should be normalized to A--0123.00000000C*
+        lcc_norm = short_lcc_to_sortable_lcc(prefix.rstrip('.'))
+        if lcc_norm:
+            result = lcc_norm.rstrip('0')
+            if '.' in prefix and prefix.endswith('0'):
+                zeros_to_add = len(prefix) - len(prefix.rstrip('0'))
+                result += '0' * zeros_to_add
+            return result.rstrip('.')
+        else:
+            return None
+
+
+def normalize_lcc_range(start, end):
+    """
+    :param str start: LCC prefix to start range
+    :param str end: LCC prefix to end range
+    :return: range with prefixes being prefixes for sortable LCCs
+    :rtype: [str, str]
+    """
+    return [
+        lcc if lcc == '*' else short_lcc_to_sortable_lcc(lcc)
+        for lcc in (start, end)
+    ]
```

Comment:
```suggestion
    return [short_lcc_to_sortable_lcc(lcc) for lcc in (start, end)]
```

Reply:
I'd like to keep `*` only inside this function, since `*` is only acceptable in the context of a range; `short_lcc_to_sortable_lcc` shouldn't accept `*` as valid.

Reply:
Resolving as won't fix.